app/forms/provider_interface/offer_wizard.rb

```diff
@@ -84,30 +84,33 @@ def next_step(step = current_step)
 
     delegate :previous_step, to: :wizard_path_history
 
-    def condition_models
-      @_condition_models ||= further_conditions.map.with_index do |further_condition, index|
-        OfferConditionField.new(id: index, text: further_condition)
+    def further_condition_models
+      @_further_condition_models ||= further_condition_attrs.map do |index, hash|
+        OfferConditionField.new(id: index.to_i, text: hash['text'], condition_id: hash['condition_id'])
```

Comment:
Maybe rename `hash` to `params` so it's more descriptive?

```suggestion
    def further_condition_models
      @_further_condition_models ||= further_condition_attrs.map do |index, params|
        OfferConditionField.new(id: index.to_i, text: params['text'], condition_id: params['condition_id'])
```